kratos/linear_solvers/amgcl_ns_solver.h

```diff
@@ -240,16 +197,16 @@ class AMGCL_NS_Solver : public LinearSolver< TSparseSpaceType,
 
         amgcl::make_solver<
```

Comment:
Save the system matrix to be used later:
```suggestion
        auto A = amgcl::adapter::zero_copy(rA.size1(), rA.index1_data().begin(), rA.index2_data().begin(), rA.value_data().begin());
        amgcl::make_solver<
```

Reply:
but do we need to make a copy? can't we make 

~~~cpp
const auto& A = ...
~~~

if a copy is actually needed i would take the occasion to copy onto the block format even for the entire matrix

Reply:
What is returned is a `std::shared_ptr` to a no-copy adapter (a csr structure holding pointers to the data owned by you). So no need to make a reference.

Reply:
The block format is not needed at this point, it will be created just for the U subsystem after the system is split four ways.